packages/react-native/src/theme/createTheme.ts

```diff
@@ -129,14 +141,14 @@ export const createTheme = (
     }) as StrictTheme['tokens']
   );
 
-  let { components } = mergedTheme;
+  let components;
```

Comment:
It's been a while since i last saw this code so might be missing something, is there a scenario where `components` should resolve to `undefined`?

Reply:
In the Theme type (both the input and output) components is optional:
https://github.com/aws-amplify/amplify-ui/blob/main/packages/react-native/src/theme/types.ts#L67
https://github.com/aws-amplify/amplify-ui/blob/main/packages/react-native/src/theme/types.ts#L95

and when our primitives reference component themes we use optional chaining:
https://github.com/aws-amplify/amplify-ui/blob/main/packages/react-native/src/primitives/Button/styles.ts#L26

so this should work right?